src/app/auth/register/register.component.html

```diff
@@ -0,0 +1,26 @@
+<form class="example-form" [formGroup]="registerForm" (ngSubmit)="onSubmit()">
+  <div *ngIf="registerForm.controls['password'].invalid && (registerForm.controls['password'].dirty || registerForm.controls['password'].touched)"
+    class="alert alert-danger">
+
+  <div *ngIf="registerForm.controls['password'].errors?.['required']">
+    password is required.
+  </div>
+  <div *ngIf="registerForm.controls['password'].errors?.['minlength']">
+    password must be at least 8 characters long.
+  </div>
+  <div *ngIf="registerForm.controls['password'].hasError('pattern')">
+    password must contain at least one special character , one number , one uppercase and one lowercase.
```

Comment:
I want you to say what the error is
is it missing a number, uppercase or lowercase or what exactly
not very generic message